### PR TITLE
[MM-20498] components/post_view/combined_system_message' to TypeScript

### PR DESCRIPTION
**Pull request by Ayanrocks**
#### Summary
 Migrated 'components/post_view/combined_system_message' module and associated tests to TypeScript  

#### Ticket Link
Fixed https://github.com/mattermost/mattermost-server/issues/15443
JIRA: https://mattermost.atlassian.net/browse/MM-20498

### CODE REVIEW

File: components/post_view/combined_system_message/combined_system_message.tsx [hunk 1]
```diff
@@ -208,7 +212,7 @@ class CombinedSystemMessage extends React.PureComponent {
         }
     }
 
-    getAllUsernames = () => {
+    getAllUsernames = ():{[p: string]: string} => {
```
Comment (devinbinnie) [inline components/post_view/combined_system_message/combined_system_message.tsx:215]: Nit: spacing around `:`

File: components/post_view/combined_system_message/index.ts [hunk 2]
```diff
@@ -2,19 +2,21 @@
...
+import {GenericAction} from 'mattermost-redux/types/actions';
 
-import CombinedSystemMessage from './combined_system_message.jsx';
+import CombinedSystemMessage, {Props} from './combined_system_message';
 
 function makeMapStateToProps() {
     const getProfilesByIdsAndUsernames = makeGetProfilesByIdsAndUsernames();
 
-    return (state, ownProps) => {
+    return (state:GlobalState, ownProps: Props) => {
```
Comment (devinbinnie) [inline components/post_view/combined_system_message/index.ts:19]: Nit: spacing around `:`

**Review by devinbinnie — APPROVED** [review]
LGTM! Just a few non-blocking comments.
Thanks @Ayanrocks!

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for contributing!

**Review by jgilliam17 — APPROVED** [review]
Thank you @Ayanrocks 
Tested, looks good to merge.
- Verified combined join/added/removed from channel system messages - as expected.
@Willyfrog Can you please help merge? Thanks. 🙂

**Pull request merged**